api/api.py

```diff
@@ -554,13 +566,34 @@ async def get_partners(
 
     return {"total_partners": total_partners, "partners": partners}
 
+@app.get(
+        "/aggregate/{state_code}",
+        name="Get aggregates by state code and territory ID",
+        response_model=AggregatesSearchResponse,
+        description="Get information about a aggregate by state code and territory ID.",
+        responses={
+            400: {"model": HTTPExceptionMessage, "description": "City not found."},
+        },)
+async def get_aggregates(territory_id: Optional[str] = Query(None, description="City's 7-digit IBGE ID."), 
+                        state_code: str = Path(..., description="City's state code.")):
+    try:
+        aggregates = app.aggregates.get_aggregates(territory_id, state_code.upper())
+        return JSONResponse(status_code=200, 
+                            content={
+                                "state_code":state_code.upper(),
+                                "territory_id":territory_id,
+                                "aggregates":aggregates}
+                            )
+    except Exception as exc:
+        return JSONResponse(status_code=404, content={"detail": str(exc)})
```

Comment:
A ideia é que o erro pode acontecer na `.get_aggregates()`, certo? Aí o `return JSONResponse()` não deveria ficar dentro do try-except
```suggestion
    try:
        aggregates = app.aggregates.get_aggregates(territory_id, state_code.upper())
    except Exception as exc:
        return JSONResponse(status_code=404, content={"detail": str(exc)})
        
    return JSONResponse(status_code=200, 
                        content={
                            "state_code":state_code.upper(),
                            "territory_id":territory_id,
                            "aggregates":aggregates}
                        )
```

Reply:
Agora é o seguinte, eu testei e não consigo gerar uma exceção esperada aqui.  Acho que o 404 deveria ser retornado quando a resposta vem vazia, como fazemos na `get_cities()` e não um try-except (já que se der exceção inesperada normalmente já dá erro 500 mesmo, que seria o mais apropriado). O que acham?

Reply:
Gente, pelo que comentei acho que podemos remover esse try-except. O resto acho que tá bão

Reply:
Deu a entender que era para implementar a verificação, mas tudo bem, iremos modificar